### PR TITLE
Update version restraints to support Symfony 4.3

### PR DESCRIPTION
**Pull request by epinxteren**
Support Symfony 4.3 with minimum effort. Most notable change is the addition of service aliases for auto wiring services.

### CODE REVIEW

File: src/DateTime/RegistrationExpirationHelper.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
 namespace Surfnet\StepupBundle\DateTime;
 
 use DateInterval;
-use DateTime;
+use DateTime as CoreDateTime;
```
Comment (epinxteren) [inline src/DateTime/RegistrationExpirationHelper.php]: In this namespace there is already a DateTime, this resolves in a error if you use it like this.

File: composer.json [hunk 2]
```diff
@@ -19,22 +19,23 @@
...
-        "symfony/framework-bundle": ">=2.7,<4",
-        "symfony/http-kernel": ">=2.7,<4",
-        "symfony/twig-bridge": ">=2.7,<4",
-        "symfony/validator": ">=2.7,<4"
+        "sensio/framework-extra-bundle": "~3 | ^5.4",
+        "surfnet/stepup-saml-bundle": "^4.0 | ^4.1",
+        "symfony/config": ">=2.7,<4|^4.3",
+        "symfony/dependency-injection": ">=2.7,<4|^4.3",
+        "symfony/form": "~2.7|~2.8|~3,4|^3.4|^4.3",
+        "symfony/framework-bundle": ">=2.7,<4|^4.3",
```
Comment (pablothedude) [inline composer.json]: Strange we need Symfony completely for a small bundle.

File: src/Request/JsonConvertibleParamConverter.php [hunk 3]
```diff
@@ -28,6 +28,7 @@
  * ParamConverter that converts JSON objects with underscore notation mapped to snake-cased, public properties of
  * classes that implement JsonConvertible.
  *
+ * @SuppressWarnings(PHPMD.MissingImport)
```
Comment (pablothedude) [inline src/Request/JsonConvertibleParamConverter.php]: Would you be so kind to add some documentation why this is used?

**Comment by MKodde** [comment]
@epinxteren Will this PR land on develop any time soon? I see you tagged some beta 4.1.x releases on this branch. 

Tomorrow I want to install some Composer dependency upgrades, and the unreleased 4.1 version might be a candidate as it will be used in the WebAuthN project, am I right?

File: composer.json (hunk 2 above)
Comment (MKodde) [reply composer.json]: Also consider adding PHP 7.x as a Composer dependency. As this bundle will now only work with newer PHP versions

File: src/DateTime/RegistrationExpirationHelper.php (hunk 1 above)
Comment (epinxteren) [reply src/DateTime/RegistrationExpirationHelper.php]: Lint issues when backslash is used, namespace should be imported

File: src/Request/JsonConvertibleParamConverter.php (hunk 3 above)
Comment (epinxteren) [reply src/Request/JsonConvertibleParamConverter.php]: Coding standerd changed, don't want to change code when it's not needed

File: composer.json [hunk 4]
```diff
@@ -19,22 +19,23 @@
...
-        "symfony/form": "~2.7|~2.8|~3,4",
-        "symfony/framework-bundle": ">=2.7,<4",
-        "symfony/http-kernel": ">=2.7,<4",
-        "symfony/twig-bridge": ">=2.7,<4",
-        "symfony/validator": ">=2.7,<4"
+        "sensio/framework-extra-bundle": "~3 | ^5.4",
+        "surfnet/stepup-saml-bundle": "^4.0 | ^4.1",
+        "symfony/config": ">=2.7,<4|^4.3",
+        "symfony/dependency-injection": ">=2.7,<4|^4.3",
+        "symfony/form": "~2.7|~2.8|~3,4|^3.4|^4.3",
```
Comment (MKodde) [inline composer.json]: Please remove the `^3.4` or the `~3.4` constraint.

Comment (MKodde) [reply composer.json]: Upon closer inspection the `~3,4` might be a typo as a comma is used.

**Comment by pablothedude** [comment]
@MKodde would you request a review when support for 7 is added? Until now all looks fine to me.

**Comment by MKodde** [comment]
@pablothedude the 7.2 build is now green as of release of https://github.com/OpenConext/Stepup-saml-bundle/releases/tag/4.1.9

File: .travis.yml [hunk 5]
```diff
@@ -1,15 +1,11 @@
...
 
-php:
-  - 5.6
-  - 7.0
+matrix:
+  include:
+    - php: 5.6
+      env: SYMFONY_VERSION="3.4.*"
+    - php: 7.2
+      env: SYMFONY_VERSION="4.3.*"
```
Comment (pablothedude) [inline .travis.yml]: Should we use 4.4 as that will become the next LTS?

Comment (MKodde) [reply .travis.yml]: I've considered this. But 4.4 is not yet released, and I think we can wait until the next update oppurtunity.

Comment (MKodde) [reply .travis.yml]: > I find it very tricky that integration tests are missing to test the used dependencies.

Please vent this on the [Pivotal ticket](https://www.pivotaltracker.com/story/show/169838423) I opened so we can address this when we are starting work on that

> I saw some dependencies which couldn't be resolved in:
> `\Surfnet\StepupBundle\Controller\ExceptionController`.
> `use Symfony\Component\Security\Core\Exception\AccessDeniedException;`
> `use Symfony\Component\Security\Core\Exception\AuthenticationException;`
> 
> See for yourself if you want to fix them as you''ll have the best feeling about this as this was encountered earlier on with the `stepup-saml-bundle`

I don't think we should fix this in the bundle. The fix would be to require symfony/symfony as a Composer requirement. But that implicitly already is the case. You would not be able to use this bundle on an application that is not of the Symfony ecosystem.

**Comment by MKodde** [comment]
> I find it very tricky that integration tests are missing to test the used dependencies.

Please vent this on the [Pivotal ticket](https://www.pivotaltracker.com/story/show/169838423) I opened so we can address this when we are starting work on that

> I saw some dependencies which couldn't be resolved in:
> `\Surfnet\StepupBundle\Controller\ExceptionController`.
> `use Symfony\Component\Security\Core\Exception\AccessDeniedException;`
> `use Symfony\Component\Security\Core\Exception\AuthenticationException;`
> 
> See for yourself if you want to fix them as you''ll have the best feeling about this as this was encountered earlier on with the `stepup-saml-bundle`

I don't think we should fix this in the bundle. The fix would be to require symfony/symfony as a Composer requirement. But that implicitly already is the case. You would not be able to use this bundle on an application that is not of the Symfony ecosystem.

**Pull request merged**